packages/design-system/src/components/WIP/Accordion/Primitive/CollapsiblePanel.tsx

```diff
@@ -0,0 +1,88 @@
+import React, {
+	cloneElement,
+	forwardRef,
+	MouseEvent,
+	ReactElement,
+	Ref,
+	useState,
+	useEffect,
+	ReactChild,
+} from 'react';
+import classNames from 'classnames';
+import CollapsiblePanelHeader from './CollapsiblePanelHeader';
+
+import styles from './CollapsiblePanel.module.scss';
+
+type CollapsiblePanelPropsType = {
+	children: ReactChild;
+	managed?: boolean;
+	expanded?: boolean;
+	index?: number;
+	title: string;
+	action?: any;
+	size?: 'S' | 'M';
+	metadata?: ReactElement[];
+	isFirst?: boolean;
+	isLast?: boolean;
+	onToggleExpanded?: (index: number) => void;
+};
+
+const CollapsiblePanel = forwardRef(
+	(
+		{
+			children,
+			managed,
+			expanded,
+			index,
+			onToggleExpanded,
+			title,
+			action,
+			size,
+			metadata,
+			isFirst = false,
+			isLast = false,
+		}: CollapsiblePanelPropsType,
+		ref: Ref<HTMLDivElement>,
+	) => {
+		const [localExpanded, setLocalExpanded] = useState(false);
+
+		useEffect(() => {
+			if (managed && expanded != localExpanded) {
+				setLocalExpanded(!!expanded);
+			}
+		}, [expanded, managed, localExpanded]);
+
+		function handleToggleExpanded() {
+			if (onToggleExpanded && managed && (index || index === 0)) {
+				onToggleExpanded(index);
+			} else {
+				setLocalExpanded(!localExpanded);
+			}
+		}
+
+		return (
+			<div
+				className={classNames(styles.panelWrapper, {
+					[styles.panelWrapper__alone]: !managed,
+					[styles.panelWrapper__first]: isFirst,
+					[styles.panelWrapper__last]: isLast,
+					[styles.panelWrapper__notLast]: managed && !isLast,
+				})}
+			>
+				<CollapsiblePanelHeader
+					expanded={localExpanded}
+					handleClick={handleToggleExpanded}
+					title={title}
+					action={action}
+					metadata={metadata}
+					size={size}
+				/>
+				{localExpanded && <div className={styles.panelContent}>{children}</div>}
```

Comment:
https://www.w3.org/WAI/ARIA/apg/example-index/accordion/accordion we're missing quite a bif ot accessibility stuff here

Reply:
I followed the guide the best I could, tell me if I missed something 🙇 .